e2e-tests/playwright/utils/helper.ts

```diff
@@ -0,0 +1,320 @@
+import { logger } from './authenticationProviders/Logger';
+import { exec } from 'child_process';
+import * as constants from './authenticationProviders/constants';
+import { expect } from '@playwright/test';
+import { kubeCLient } from './k8sHelper';
+import { V1ConfigMap, V1Secret } from '@kubernetes/client-node';
+
+export const k8sClient = new kubeCLient();
+
+export async function runShellCmd(command: string) {
+  return new Promise<string>((resolve, reject) => {
+    logger.info(`Executing command ${command}`);
+    const process = exec(command);
+    let result: string;
+    process.stdout.on('data', data => {
+      result = data;
+    });
+    process.stderr.on('data', data => {
+      logger.log({
+        level: 'error',
+        message: `Error occurred while running command`,
+        dump: data,
+      });
+      reject();
```

Comment:
If the program writes to stderr, it doesn't mean that there was an error.
Stderr is sometimes used to output debugging or tracing information. 

The right way to determine if there was an error while executing the program is to exit the code of the program. 



 I would suggest something like this:
 
 ```javascript
export async function runShellCmd(command: string) {
  return new Promise<string>((resolve, reject) => {
    logger.info(`Executing command ${command}`);
    const process = exec(command);
    let result: string;
    process.stdout.on('data', data => {
      logger.log({
        level: 'debug',
        message: `Stdout output:`,
        dump: data,
      });
      result = data;
    });
    process.stderr.on('data', data => {
      logger.log({
        level: 'debug',
        message: `Stderr output:`,
        dump: data,
      });
    });
    process.on('exit', code => {
      if (code == 0) {
        resolve(result)
      } else {
        reject();
        throw Error('Error executing shell command');
      }
    });
  });
```

Reply:
Updated to check the exit code.